examples/solvers/adupdates_tomography.py

```diff
@@ -0,0 +1,151 @@
+r"""Total-variation regularized tomography example using the adupdates solver.
+
+This example solves a linear inverse problem of the form :math:`Ax = y`, where
+:math:`x \in \mathbb{R}^n` is the (unknown) original image to be reconstructed,
+:math:`y \in \mathbb{R}^m` is the noisy data and :math:`A \in \mathbb{R}^{m
+\times n}` is the measurement matrix describing the discretized physical model.
+
+To solve this problem, we first split the measurement matrix and the data into
+:math:`m` parts and solve the least squares problem
+
+..math::
+    \min_{x\in\mathbb{R}^n} D(x) := \sum_{j = 1}^m \| A_j x - y_j \|^2.
+
+To regularize, we add terms for the total variation and for a functional which
+guarantees that the solution will be pointwise nonnegative. The variational
+regularization problem we are solving is therefore
+
+..math::
+    \min_{x\in \mathbb{R}^n_+} D(x) + \sum_{i = 1}^d (\| \partial_{i, 1} x \|_1
+    + \| \partial_{i, 2} x \|_1.
+
+Here, :math:`\partial_{i, 1}` and :math:`\partial_{i, 2}` contain the even and
+odd components, respectively, of the discretized :math:`i`th partial
+derivative, and :math:`d` is the dimension of the tomography problem. In this
+example, :math:`d = 2`. We solve the problem with the alternating dual updates
+method. For further details, see
+`[MF2015] <http://ieeexplore.ieee.org/document/7271047/>`_.
+
+References
+----------
+[MF2015] McGaffin, M G, and Fessler, J A. *Alternating dual updates
+algorithm for X-ray CT reconstruction on the GPU*. IEEE Transactions
+on Computational Imaging, 1.3 (2015), pp 186--199.
+"""
+
+import numpy as np
+import odl
+
+# Some constants we shall use in the sequel to avoid magic numbers.
+DIMENSIONS = 40.0     # Dimensions of reconstruction.
+RESOLUTION = 1024     # Discretization level of reconstruction space.
+SPLIT_METHOD = 'interlaced'  # How to split the data?
+SPLIT_NUMBER = 20     # How many pieces of data?
+NOISE_LEVEL = 5.0     # How much noise to add to the sinograms
+DATA_FIT_WEIGHT = 1.0  # Regularization parameter (coefficient of the data fit)
+EPSILON = 1e-6        # make sure that we do not divide by zero by adding this
+STEPSIZE = 1.         # Stepsize of the outer iterations
+NITER = 5             # Number of iterations
+RANDOM = True         # Choose the oder of the inner iterations at random?
```

Comment:
While I appreciate the flexibility that comes with this parametrization, I think that this level of flexibility tends to scare off people. There are just too many choices. The script misses its purpose as an example (aimed at simplicity and comprehensibility) and rather comes as a command line utility in disguise (aimed at flexibility and full-featuredness).

On top of that, not all of the parameters are really needed. Some are just "for implementer's convenience" and could be handled in a better way. Examples:

- `DIMENSIONS`: Not needed. You can simply hardcode the extent of the reconstruction space and retrieve it later when you need it. That still gives you exactly one place to change it. (Apart from that, the name is slightly wrong: `HALF_DIMENSIONS` would be more accurate.)
- `RESOLUTION`: Apart from the fact that most people would not think of resolution in this way, you don't actually need it as a module constant, just like `DIMENSIONS`.

... and so on.

Ultimately, I think the only interesting knobs that I imagine somebody playing with this script would like to turn are related to the problem splitting, i.e., `SPLIT_NUMBER` AND `SPLIT_METHOD`, perhaps `RANDOM`. For the others: just do it as in the other examples and inline them.